torchao/float8/inference.py

```diff
@@ -7,246 +7,32 @@
-        """This functions converts the weight to a Float8Tensor and sets its requires_grad to False.
-
-        Args:
-            dtype: The dtype to quantize the weight to. Default is e4m3_dtype.
-
-        Note:
-            This function is typically called during inference to quantize the weight once since
-            the weight is not updated during inference.
-
-        """
-        assert not isinstance(
-            self.weight, Float8Tensor
-        ), "Weight has already been quantized, cannot quantize again."
-        scale = tensor_to_scale(self.weight, dtype)
-        quantized_weight = hp_tensor_and_scale_to_float8(
-            self.weight,
-            scale,
-            dtype,
-            self.linear_mm_config,
-            GemmInputRole.WEIGHT,
-        )
-        self.weight = nn.Parameter(quantized_weight)
-        self.weight.requires_grad = False
-
-    def set_weight_and_bias(
-        self, weight: torch.nn.Parameter, bias: Optional[torch.nn.Parameter]
-    ):
-        self.weight = weight
-        self.bias = bias
 
-    @classmethod
-    def from_float(
-        cls, module: nn.Module, quant_config: QuantConfig, use_fast_accum: bool
-    ) -> "Float8InferenceLinear":
-        """
-        Create an nn.Linear with fp8 compute from another nn.Linear
-
-        Args:
-            module (torch.nn.Linear): nn.Linear to convert
-            quant_config (QuantConfig): Configuration for the weight and activation casting
-        """
-        forward_config = ScaledMMConfig(
-            False, use_fast_accum, pad_inner_dim=quant_config.pad_inner_dim
-        )
-        linear_mm_config = LinearMMConfig(
-            forward_config, forward_config, forward_config
-        )
-        linear = cls(
-            quant_config,
-            linear_mm_config,
-            module.in_features,
-            module.out_features,
-            False,
-            device=torch.device("meta"),
-        )
-        linear.set_weight_and_bias(module.weight, module.bias)
-        linear.quantize_weight()
-        return linear
-
-
-def cast_to_float8_e4m3_inference(
-    input_tensor: torch.Tensor,
-    linear_mm_config: LinearMMConfig,
-    reduce_amax: bool = False,
-    static_quantization_scale: Optional[torch.Tensor] = None,
-) -> Float8Tensor:
-    """Casts an input tensor to the Float8 (e4m3fn*)
-
-    Args:
-        input_tensor: The input tensor to be cast.
-        linear_mm_config: Configuration settings for the matrix multiplication
-        reduce_amax: Whether to reduce the amax (absolute maximum) among the local distributed group.
-        static_quantization_scale: Optional tensor specifying the scale for activation. Default is None.
-
-    Returns:
-        Float8Tensor: The input tensor cast to Float8 (e4m3fn) format.
-
-    Note:
-        If the input tensor is already in Float8 format, it is returned as is without re-casting.
+class Float8MMConfig(NamedTuple):
```

Comment:
can we make the names `ScaledMMConfig` and `Float8MMConfig` not be confusing?  Ideally it should be clear why there are two objects and in which context the user should use which object

Reply:
Sure, I was going to do ScaledMMConfigInference but that is even too verbose for me, any suggestions?

Reply:
for training I went with a user facing `Float8GemmConfig` which is a dataclass and easy to explain (IMO), the way the data gets passed around is not user facing so details of `ScaledMMConfig` don't matter as much

tbh making `ScaledMMConfig` public and reusable between training + inference sounds right to me, if you really want this to be public without a dataclass wrapper

Reply:
It can not be a dataclass today because that wont work with compile, Lazos has a PR to make frozen dataclasses proxyable  but until then it has to be a named tuple. 

I love dataclasses but I dont understand why they are more understandable then a namedtuple, Is the problem that this name is similar to other names?